### PR TITLE
[REVIEW] Add BlazingSQL version of query 15

### PR DESCRIPTION
**Pull request by Christian8491**
This PR adds the version of "query" 15 using `BlazingSQL`.

### CODE REVIEW

**Comment by Christian8491** [comment]
It shows **Correctness Assertion True** at scale factor **1K**, using **dgx201**

```
(tpcx-bb) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q15$ python tpcx_bb_query_15_sql.py --data_dir=/raid/tpcx-bb/sf1000/parquet_2gb/ --verify_results --verify_dir=/mnt/weka/nicholasb/tpcx-bb/validation/sf1000/ --cluster_host=10.150.162.27
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx-bb/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
Standard ETL Query
/home/christianc/miniconda3/envs/tpcx-bb/lib/python3.7/site-packages/pandas/util/__init__.py:12: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing
Correctness Assertion True
Must pass a sheet and tab name to use Google Sheets automation
```

The "query" finishes successfully at scale factor **10K**

```
(tpcx-bb) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q15$ python tpcx_bb_query_15_sql.py --data_dir=/mnt/weka/tpcx-bb/sf10000/parquet_3gb/ --cluster_host=10.150.162.27
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx-bb/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
Must pass a sheet and tab name to use Google Sheets automation
```

Do you think this PR should change to **REVIEW**?

File: tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# limitations under the License.
+#
+
+import sys
+
+
+from blazingsql import BlazingContext
+from xbb_tools.cluster_startup import attach_to_cluster
+from dask_cuda import LocalCUDACluster
+from dask.distributed import Client
```
Comment (beckernick) [inline tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py:24]: unused imports

File: tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+
```
Comment (beckernick) [inline tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py:19]:
```suggestion
```

File: tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                SELECT
+                    i.i_category_id AS cat, 
+                    s.ss_sold_date_sk AS x,
+                    CAST(SUM(s.ss_net_paid) AS DOUBLE) AS y,
+                    CAST(s.ss_sold_date_sk * SUM(s.ss_net_paid) AS DOUBLE) AS xy,
+                    CAST(s.ss_sold_date_sk * s.ss_sold_date_sk AS DOUBLE) AS xx
+                FROM store_sales s
+                INNER JOIN item i ON s.ss_item_sk = i.i_item_sk
+                INNER JOIN date_dim d ON s.ss_sold_date_sk = d.d_date_sk
+                WHERE s.ss_store_sk = 10
```
Comment (beckernick) [inline tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py:68]: Would be nice to factor out the `10` into a variable with a name matching the original Spark for clarity. You can probably use the ones from the pure Dask query

File: tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py (hunk 1 above)
Comment (Christian8491) [reply tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py]: Unused imports removed.

File: tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py (hunk 2 above)
Comment (Christian8491) [reply tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py:19]: Line 19 removed.

File: tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py (hunk 3 above)
Comment (Christian8491) [reply tpcx_bb/queries/q15/tpcx_bb_query_15_sql.py]: Additional to the  `10` value (`q15_store_sk`),  the variables `q15_startDate` and  `q15_endDate` also were considered as original Spark version.

**Comment by Christian8491** [comment]
After applying these changes, I just tested again at sf **1K** and got the **Correctness Assertion True** message again using the **dgx201**, so updating this PR to **REVIEW**.

**Pull request merged**